crates/util/python-utils/src/bridge/types.rs

```diff
@@ -0,0 +1,166 @@
+    /// user in their `OP_RETURN` output.
+    pub x_only_public_key: XOnlyPublicKey,
+
+    /// The original script_pubkey in the Deposit Request Transaction (DRT) output used to sanity
+    /// check computation internally i.e., whether the known information (n/n script spend path,
+    /// [`static@UNSPENDABLE_INTERNAL_KEY`]) + the [`Self::take_back_leaf_hash`] yields the same
+    /// P2TR address.
+    pub original_script_pubkey: ScriptBuf,
+}
+
+#[pymethods]
+impl DepositRequestData {
+    #[getter]
+    fn stake_index(&self) -> u32 {
+        self.stake_index
+    }
+
+    #[getter]
+    fn ee_address(&self) -> Vec<u8> {
+        self.ee_address.clone()
+    }
+
+    #[getter]
+    fn total_amount(&self) -> u64 {
+        self.total_amount.to_sat()
+    }
+
+    #[getter]
+    fn x_only_public_key(&self) -> String {
+        self.x_only_public_key.to_string()
+    }
+}
+
+/// Withdrawal fulfillment transaction metadata
+#[derive(Debug, Clone)]
+pub(crate) struct WithdrawalMetadata {
+    /// The tag used to mark the withdrawal metadata transaction
+    pub tag: [u8; 4],
+
+    /// The index of the operator
+    pub operator_idx: u32,
+
+    /// The index of the deposit
+    pub deposit_idx: u32,
+
+    /// The txid of the deposit UTXO
+    pub deposit_txid: Txid,
+}
+
+// WithdrawalMetadata implementations
+impl WithdrawalMetadata {
+    pub(crate) fn new(
+        tag: [u8; 4],
+        operator_idx: u32,
+        deposit_idx: u32,
+        deposit_txid: Txid,
+    ) -> Self {
+        Self {
+            tag,
+            operator_idx,
+            deposit_idx,
+            deposit_txid,
+        }
+    }
+
+    pub(crate) fn op_return_data(&self) -> Vec<u8> {
+        let op_id_prefix: [u8; 4] = self.operator_idx.to_be_bytes();
+        let deposit_id_prefix: [u8; 4] = self.deposit_idx.to_be_bytes();
+        let deposit_txid_data = consensus::encode::serialize(&self.deposit_txid);
+        [
+            &self.tag[..],
+            &op_id_prefix[..],
+            &deposit_id_prefix[..],
+            &deposit_txid_data[..],
+        ]
+        .concat()
+    }
+
+    pub(crate) fn op_return_script(&self) -> PushBytesBuf {
+        let data = self.op_return_data();
+        let mut push_data = PushBytesBuf::new();
+
+        push_data
+            .extend_from_slice(&data)
+            .expect("metadata should be within push data limits");
+
+        push_data
+    }
+}
+
+/// Deposit transaction metadata for OP_RETURN
+#[derive(Debug, Clone)]
+pub(crate) struct DepositTxMetadata {
+    pub stake_index: u32,
+    pub ee_address: Vec<u8>,
+    pub takeback_hash: TapNodeHash,
+    pub input_amount: Amount,
+}
+
+/// Keys
+#[pyclass]
+#[derive(Debug, Clone)]
+pub struct OperatorKeys {
+    keys: Vec<Keypair>,
+}
+
+#[pymethods]
+impl OperatorKeys {
+    #[new]
+    pub fn new(keys: Vec<String>) -> PyResult<Self> {
+        Ok(Self {
+            keys: parse_operator_keys(&keys)?,
+        })
```

Comment:
This should take a Python `bytes` or even an already parsed `Keypair` type.  Keys should be parsed from strings *as early as possible*.  The Ethereum JS ecosystem is really bad about this.